geode-core/src/distributedTest/java/org/apache/geode/cache30/ReconnectWithCacheXMLDUnitTest.java

```diff
@@ -55,10 +61,22 @@
   @Rule
   public DistributedRestoreSystemProperties restoreSystemProperties =
       new DistributedRestoreSystemProperties();
+  private int locatorPort;
 
   @Override
   public final void postSetUp() {
     oldPropertySetting = System.setProperty(xmlProperty, "true");
+    // stress testing needs this so that join attempts don't give up too soon
+    Invoke.invokeInEveryVM(() -> System.setProperty("p2p.joinTimeout", "120000"));
+    // reconnect tests should create their own locator so as to not impact other tests
+    VM locatorVM = VM.getVM(0);
+    final int port = locatorVM.invoke(() -> {
+      System.setProperty(BYPASS_DISCOVERY_PROPERTY, "true");
+      System.setProperty(DistributionConfig.GEMFIRE_PREFIX + "member-weight", "100");
```

Comment:
Can't the test set member-weight through the gemfire properties? Setting a system property seems yuckier to do in a test, because it will affect all future tests in this JVM.

Reply:
https://issues.apache.org/jira/browse/GEODE-7602

Reply:
The test uses DistributedRestoreSystemProperties to reset system properties when it finishes.